src/primitives.jl

```diff
@@ -160,23 +160,50 @@ Return a vector containing only the non-null entries of `X`,
 unwrapping `Nullable` entries. A copy is always returned, even when
 `X` does not contain any null values.
 """
-function dropnull(X::AbstractVector)                 # -> AbstractVector
-    Y = filter(x->!_isnull(x), X)
-    res = similar(Y)
-    for i in eachindex(Y, res)
-        res[i] = isa(Y[i], Nullable) ? Y[i].value : Y[i]
+function dropnull{T}(X::AbstractVector{T})                  # -> AbstractVector
+    if !(Nullable <: T) && !(T <: Nullable)
+        return copy(X)
+    else
+        Y = filter(x->!_isnull(x), X)
+        res = similar(Y, eltype(T))
+        for i in eachindex(Y, res)
+            @inbounds res[i] = isa(Y[i], Nullable) ? Y[i].value : Y[i]
+        end
+        res
```

Comment:
Maybe make this `return res` for consistency with the first branch, and for clarity since that's not the last line in the function.